src/modules/custom.rs

```diff
@@ -120,23 +125,30 @@ fn shell_command(cmd: &str, shell: Option<&str>) -> Option<Output> {
 /// Attempt to run the given command in a shell by passing it as `stdin` to `get_shell()`,
 /// or by invoking cmd.exe /C.
 #[cfg(windows)]
-fn shell_command(cmd: &str, shell: Option<&str>) -> Option<Output> {
-    let shell = if let Some(shell) = shell {
-        Some(std::borrow::Cow::Borrowed(shell))
+fn shell_command(cmd: &str, shell_args: &[&str]) -> Option<Output> {
+    let (shell, shell_args) = if !shell_args.is_empty() {
+        (
+            Some(std::borrow::Cow::Borrowed(shell_args[0])),
+            &shell_args[1..],
+        )
     } else if let Ok(env_shell) = std::env::var("STARSHIP_SHELL") {
-        Some(std::borrow::Cow::Owned(env_shell))
+        (Some(std::borrow::Cow::Owned(env_shell)), shell_args)
```

Comment:
The code here makes me believe that it's possible to specify shell arguments without specifying a shell and relying on the `STARSHIP_SHELL` environment variable -- I think that's great as we can define custom parameters such as loading profile if desired. But from the first and previous condition, it seems that as soon as the `shell` configuration variable is set, the first parameters is assumed to be the shell. Hence we'll never have valid `shell_args` here. Is this on purpose?

I might be missing something.. can you show me an example of defining shell parameters without defining the shell?

Reply:
I did it to trick the type checker: in these branches `shell_args` are empty, but they have the right lifetime to be returned. After some digging, `&[]` has a static lifetime so that works too. Pushing this change for clarity.

Reply:
Okay, so it's more of a pragmatic reason.

I assume then that we'll have to override the shell by defining it as a first parameter to also provide the arguments? That make sense, thinking about it again. The user couldn't be 100% sure which shell that Starship will choose and overriding the shell will lead to the path of least surprises. 

Thank you!